### PR TITLE
cmd personal logistics for spidertron

### PR DESCRIPTION
**Pull request by PHIDIAS0303**

### CODE REVIEW

File: modules/data/personal-logistic.lua [hunk 1]
```diff
@@ -63,12 +73,32 @@ local function pl(player, amount)
...
 
-Commands.new_command('personal-logistic', 'Set Personal Logistic (0 to cancel all)')
+Commands.new_command('personal-logistic', 'Set Personal Logistic (0 to cancel all) (Select spidertron to edit spidertron)')
 :add_param('amount', 'integer-range', 0, 10)
 :add_alias('pl')
 :register(function(player, amount)
     if player.force.technologies['logistic-robotics'].researched then
-        pl(player, amount / 10)
+        if player.selected ~= nil then
+            if player.selected.name ~= nil then
```
Comment (Cooldude2606) [inline modules/data/personal-logistic.lua:82]: `name` is never nil. This check is not required.

File: modules/data/personal-logistic.lua [hunk 2]
```diff
@@ -1,30 +1,40 @@
...
+            c = target.clear_personal_logistic_slot
+            s = target.set_personal_logistic_slot
 
-    for k, v in pairs(config.request) do
+            for k, v in pairs(config.request) do
+                c(config.start + v.key)
+                s(config.start + v.key, {name=k, min=0, max=0})
+            end
+
+            return
```
Comment (Cooldude2606) [inline modules/data/personal-logistic.lua:25]: There is no benifit to include "clear" in this function. Either make it into its own function or place the for loop directly in the command body.
After you have done this there will be no difference between "player" (`p`) and "selected" (`s`) so you can remove type from the arguments and set `c` and `s` unconditionally.

**Review by Cooldude2606 — CHANGES_REQUESTED** [review]
Small improvements can be made, but functionaly is all correct.

**Review by Cooldude2606 — APPROVED** [review]
All looks good. You can still have the second command if you want, my change request was that it should be in its own function.

**Comment by PHIDIAS0303** [comment]
it is not really needed for the 2nd cmd i think

**Pull request merged**